client/controllers/incidentReports/incidentForm.coffee

```diff
@@ -124,6 +127,31 @@ Template.incidentForm.helpers
 
   diseaseOptionsFn: -> diseaseOptionsFn
 
+  speciesOptionsFn: ->
+    return (params, callback) ->
+      term = params.term?.trim()
+      if not term
+        return callback(results: [])
+      Meteor.call 'searchSpeciesNames', term, (error, results)->
+        if error
+          notify('error', error.reason)
+        console.log results
+        callback(
+          results: results.map((item)->
+            text = item.completeName
+            if (new RegExp(term, "i")).test(item.vernacularName)
+              text = item.vernacularName + " | " + item.completeName
+            {
+              id: 'tsn:' + item.tsn
+              text: text
+              item: item
+            }
+          ).concat([
+            id: "userSpecifiedSpecies:#{term}"
+            text: "Other Species: #{term}"
+          ])
+        )
+
```

Comment:
There may be a few areas where we could update the code style to stay consistent with [our guide](https://github.com/ecohealthalliance/tater/wiki/Coffeescript-Style-Guide).
Like line 135 (adding space before `->`) and maybe removing some of the parens?